tomcat/src/main/tomcat-base/conf/server.xml

```diff
@@ -17,6 +17,7 @@
   See the License for the specific language governing permissions and
   limitations under the License.
 -->
+<!DOCTYPE server-xml [<!ENTITY gcp-configuration SYSTEM "gcp.xml">]>
```

Comment:
If `gcp.xml` is empty then nothing is added to the configuration (gcp valves). If `gcp.xml` is missing then an error is thrown when the server is started.